elastic/kibana
# [ILM] Fix missing ILM data on hidden indices

## Summary

Fixes https://github.com/elastic/kibana/issues/88441

This PR fixes the ILM data enricher to fetch the ILM data for hidden indices, it also sets the response from ES to only return ILM data for indices that are managed by ILM.

To test this fix, follow these steps:
1. Add a sample data set
2. Navigate to Discover and click "share" to create a CSV report
3. Navigate to ILM and toggle "include managed policies"
4. Find the ILM policy "kibana-reporting" and click the link for linked indices
5. In Index Management check that the linked index ".reporting*" is found in the table and has the ILM data attached to it

### Screenshot 
<img width="1118" alt="Screenshot 2024-02-02 at 18 14 50" src="https://github.com/elastic/kibana/assets/6585477/e1f7c8d2-69f1-4925-a613-6b0b0a6a72d5">



### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
